### ISSUE TITLE
Pre-provisioned nodes cause errors in cluster-autoscaler

### ISSUE THREAD

**Issue by jcrowthe** [kubernetes/autoscaler#756]
When using cluster-autoscaler with an Azure VM Scale Set that already has a node in it, I receive the following incessantly after a ScaleUp and ScaleDown cycle:

```
I0328 23:35:35.620211       1 static_autoscaler.go:184] 1 unregistered nodes present
I0328 23:35:35.690141       1 utils.go:330] Removing unregistered node
I0328 23:35:35.690224       1 azure_scale_set.go:273] Deleting vmss instances [%!q(*azure.azureRef=&{})]
E0328 23:35:35.690245       1 azure_scale_set.go:293] getLastSegment failed with error: identifier '/' not found in resource name ""
W0328 23:35:35.690253       1 utils.go:353] Failed to remove node : identifier '/' not found in resource name ""
E0328 23:35:35.690270       1 static_autoscaler.go:191] Failed to remove unregistered nodes: identifier '/' not found in resource name ""
```

In looking at successful ScaleDown operations, the third line from above should appear more like `I0328 22:50:57.080662       1 azure_scale_set.go:273] Deleting vmss instances [%!q(*azure.azureRef=&{azure:///subscriptions/<sub>/resourceGroups/aaakube56/providers/Microsoft.Compute/virtualMachineScaleSets/vmss-agent-worker-aaakube56/virtualMachines/9})]`

This ID does appear properly on all nodes under their ProviderID property, such as:
```
$ k describe node vmss-agent-worker-aaakube56000000
...
ProviderID:                  azure:///subscriptions/<sub>/resourceGroups/aaakube56/providers/Microsoft.Compute/virtualMachineScaleSets/vmss-agent-worker-aaakube56/virtualMachines/0
```

**Comment by feiskyer** [comment]
@jcrowthe It seems azure_scale_set got an empty instance, let me check what's wrong

**Comment by feiskyer** [comment]
@jcrowthe Confirmed this is a bug in azure provider, will file a PR to fix this

**Comment by feiskyer** [comment]
@sozercan Sorry, this is not included in v1.2.0. It should be included in v1.2.1
